src/components/TodoForm/TodoForm.tsx

```diff
@@ -0,0 +1,105 @@
+import { useState } from 'react';
+import usersFromServer from '../../api/users';
+import { Todo } from '../../types/Todo';
+import { getUserById } from '../../servises/user';
+
+type Props = {
+  onSubmit: (todo: Todo) => void;
+};
+
+export const TodoForm: React.FC<Props> = ({ onSubmit }) => {
+  const [count, setCount] = useState(0);
+
+  const [title, setTitle] = useState('');
+  const [userId, setUserId] = useState(0);
+
+  const [hasTitleError, setHasTitleError] = useState(false);
+  const [hasUserIdError, setHasUserIdError] = useState(false);
+
+  const handleTitleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const pattern = /[^A-Za-z0-9А-я\s]+/;
+
+    if (!pattern.test(event.target.value)) {
+      setTitle(event.target.value);
+      setHasTitleError(false);
+    }
+  };
+
+  const handleUserIdChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    setUserId(+event.target.value);
+    setHasUserIdError(false);
+  };
+
+  const reset = () => {
+    setTitle('');
+    setUserId(0);
+    setCount(count + 1);
+  };
+
+  const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    setHasTitleError(!title);
+    setHasUserIdError(!userId);
+
+    if (!title || !userId) {
```

Comment:
```suggestion
    if (!title.trim() || !userId) {
```
to disable creating todo only with spaces
<img width="217" alt="image" src="https://github.com/user-attachments/assets/a039566d-15ba-4e03-974f-816bc82f477b">